gcsfs/core.py

```diff
@@ -589,11 +589,22 @@ async def _get_object(self, path):
             # listing.
             raise FileNotFoundError(path)
 
-        result = self._process_object(
-            bucket, await self._call("GET", "b/{}/o/{}", bucket, key, json_out=True)
-        )
-
-        return result
+        res = None
+        # Work around various permission settings. Prefer an object get (storage.objects.get), but
+        # fall back to a bucket list + filter to object name (storage.objects.list).
+        try:
+            res = await self._call("GET", "b/{}/o/{}", bucket, key, json_out=True)
+        except OSError as e:
+            if not str(e).startswith("Forbidden"):
+                raise
+            resp = await self._call("GET", "b/{}/o/", bucket, prefix=key, json_out=True)
```

Comment:
Should we have limit=1 here (I think GCS allows this) - the correct entry should always be first alphabetically.